### PR TITLE
More stuff

### LINKED ISSUE #972: Failed to write settings when MO ini save in UTF8-BOM
## The problem:
MO shows the error if ini was saved in UTF8-BOM even if it is file with identical parameters

## To Reproduce:
Save ModOrganizer.ini in UTF8-BOM and run MO

## Environment:

* Mod Organizer Version that exhibits the issue: 2.2.2.1
* Last Mod Organizer Verison that did not exhibit the issue (if applicable): 2.2.1
* Desktop OS/version used to run Mod Organizer: Win7

## Details:
Some other utilities with ini parsers which can save MO ini in UTF8-BOM but v2.2.2 started to throw error about it. Ofcourse nothing happens after this and the ini just will be resaved but each time if the ini will be saved externally MO will show the error. With v2.2.1 was no any kind of problem. It breaks nothing but each time shows the error about ini is broken even if it is not broken.

### LINKED ISSUE #995: Crash when mods are removed from the filtered list
## The problem:
When a filter is active in the mod list, making changes to mods that would remove mods from the filter is liable to make MO crash.

## To Reproduce:

1. Create a bunch of empty mods.
2. Filter by "Not <Has valid game data>".
3. Select all the mods.
4.  Right-click and select "ignore missing data".
5. Wait for the crash.

## Environment:

* Mod Organizer Version that exhibits the issue: 2.2.2.1
* Last Mod Organizer Verison that did not exhibit the issue (if applicable): ???
* Desktop OS/version used to run Mod Organizer: Windows 10, version 1909 (18363.592)

## Logs: 
[log + crashdump.zip](https://github.com/ModOrganizer2/modorganizer/files/4186195/log.%2B.crashdump.zip)

Note: Log mentions RTSS.  Disabling RTSS does not solve the issue.

### LINKED ISSUE #1020: Actions on mods only work on every other mod
MO 2.2.2.1
Sorting mod list by flags
Select multiple mods that have the same flag (e.g., no valid game data or alternate game source)
Select the thing in the context menu to make that flag go away
Notice that you have to do it multiple times to make the flag go away on everything

Example: 
Multiple ignores

### LINKED ISSUE #1071: Mods wont sort without internet
Mods wont sort without internet. Most of the time I will not have internet connection. This makes MO2 unusable and obsolete in every way.

### LINKED ISSUE #1156: Option to define browser path and exe for opening all URL content
As per title says, it be nice to have a box with command to open browser when trying to view URL content from MO2.

This is needed sometimes as MO2 can incorrectly detect the wrong browser, or no browser (and freeze), or in situation where you want it to use a specific browser or set of browser opening commands (incognito mode, new browser windows etc.. all done via direct command lines).

Additionally this would help solve issues with stuff like proton and wine which for some reason often fail to open the browser correctly from MO2, likely because the command to do so is being called wrong. 

This additional option should fix all that.

### LINKED ISSUE #1206: Improve text filter for downloads tab (filter by Id etc).
Could the filter for the downloads include a feature the mod side uses. You can filter by nexus id by typing in the filter box, it works for the mod list but not the download list. I'm using MO 2.3.2 and have nexus id show for both lists. If this could be added I'm sure a lot of people like me would appreciated it ! :)  Thanks!

### LINKED ISSUE #1320: API quota continues to be used in offline mode
## To Reproduce:

1. Launch MO2.
2. Log into a premium nexus account.
3. Ensure 'Offline Mode' is checked.
4. File -> Install Mod...
5. Select a mod file from the local filesystem.
6. Complete the installation process of the mod.
7. Observe API quota reduction and info log output.

## Environment:

* Mod Organizer Version that exhibits the issue: v2.3.2
* Last Mod Organizer Version that did not exhibit the issue (if applicable): unsure
* Desktop OS/version used to run Mod Organizer: Windows 10 Pro v1903 OS build 18362.657

## Link to Mod Organizer logs:

Logs kind of speak for themselves:

```
[2020-12-12 10:12:40.032 I] Checking updates for 1 mods...
[2020-12-12 10:13:24.294 I] Checking updates for 1 mods...
[2020-12-12 10:13:49.605 I] Checking updates for 1 mods...
[2020-12-12 10:14:07.382 I] Checking updates for 1 mods...
[2020-12-12 10:16:48.927 I] Checking updates for 1 mods...
```

One would expect no updates to be checked in Offline Mode.

### PR DESCRIPTION
**Pull request by isanae**
- Allow for selecting another instance when the game/variant for the current one isn't found, instead of being stuck in a loop
- Various fixes for offline mode (update check when installing, login on startup, loot master list, etc.) (fixes #1320, fixes #1071)
- Downloads:
   - Split `DownloadsTab` out of `MainWindow`
   - Now using `FilterWidget` (fixes #1206)
- Fixed `FilterWidget::m_useDelay` being uninitialized in uibase so it's `false`. Removed all `setUpdateDelay(false)` because they're now redundant, only the data tab needs it.
- Added custom browser command. Changed a bunch of places that were using `QDesktopServices` directory to use `shell::Open()` from uibase, which has a new global `SetUrlHandler()` (fixes #1156)
- Fixed problems with ignore data/mark converted in the mod list when the list is filtered so the mods disappear (fixes #1020, fixes #995)
- Fixed warning for INI files with a utf-8 bom (fixes #972)
- Saves tab:
   - Split `SavesTab` out of `MainWindow`
   - Added delete shortcut for saves game (part of #930)
   - Added "open in explorer" context menu item

### CODE REVIEW

File: src/mainwindow.cpp [hunk 1]
```diff
@@ -3118,41 +3118,51 @@ void MainWindow::displayModInformation(int row, ModInfoTabIDs tabID)
...
       ModInfo::Ptr info = ModInfo::getByIndex(row_idx);
       info->markConverted(true);
-      connect(this, SIGNAL(modListDataChanged(QModelIndex, QModelIndex)), m_OrganizerCore.modList(), SIGNAL(dataChanged(QModelIndex, QModelIndex)));
-      emit modListDataChanged(m_OrganizerCore.modList()->index(row_idx, 0), m_OrganizerCore.modList()->index(row_idx, m_OrganizerCore.modList()->columnCount() - 1));
+      changed.push_back(info);
+    }
+
+    for (auto&& m : changed) {
+      int row_idx = ModInfo::getIndex(m->internalName());
+      m_OrganizerCore.modList()->notifyChange(row_idx);
```
Comment (Al12rs) [inline src/mainwindow.cpp:3164]: If selection is sequential wouldn't it be better to use the notifyChange(startRow, endRow) ?  This is the kind of thing that causes stuff to refresh multiple times in a row

Comment (isanae) [reply src/mainwindow.cpp:3164]: Maybe. When the list is filtered, the rows can disappeared, and I'm not sure when or where, so I don't know whether a range would work. I'm not too keen on trying it.

Comment (Al12rs) [reply src/mainwindow.cpp:3015]: Another thing is that by using notifyChange you are forcing a recalculation of the conflicts of the mods in question, while before it was emitting dataChanged() with the weird connection which (I don't think) forced the conflicts refresh.

Comment (isanae) [reply src/mainwindow.cpp:3015]: All `notifyChange()` does is fire `dataChanged()`, there's no change afaict.

File: src/downloadlist.cpp [hunk 2]
```diff
@@ -203,3 +208,74 @@ void DownloadList::update(int row)
   else
     log::error("invalid row {} in download list, update failed", row);
 }
+
+bool DownloadList::lessThan(const QModelIndex &left, const QModelIndex &right)
```
Comment (Holt59) [inline src/downloadlist.cpp:212]: Could this be named something else? I know the name is understandable but it's also the named use by the proxy model function so it took me some time to understand that this is not actually directly called and is simply used as the predicate for the filter widget.

File: src/mainwindow.cpp (hunk 1 above)
Comment (Al12rs) [reply src/mainwindow.cpp:3015]: You are right I read notifychange wrong, it does the conflict checks again only if it's passed -1.

**Review by Holt59 — APPROVED** [review]

**Pull request merged**